### PR TITLE
configure external TURN server

### PR DESCRIPTION
**Pull request by tbutter**
Add 3 env variables to allow configuring an external TURN server

### CODE REVIEW

File: prosody/rootfs/defaults/conf.d/jitsi-meet.cfg.lua [hunk 1]
```diff
@@ -22,6 +22,22 @@ http_default_host = "{{ .Env.XMPP_DOMAIN }}"
...
+{{ end }}
+
+{{ if .Env.TURN_HOST }}
+
+turncredentials_host = "{{ .Env.TURN_HOST }}";
+turncredentials_port = "{{ $TURN_PORT }}";
+
+turncredentials = {
+  { type = "turn", host = "{{ .Env.TURN_HOST }}", port = "{{ $TURN_PORT }}", transport = "tcp" },
+  { type = "turns", host = "{{ .Env.TURN_HOST }}", port = "{{ $TURN_PORT }}", transport = "tcp" }
```
Comment (saghul) [inline prosody/rootfs/defaults/conf.d/jitsi-meet.cfg.lua:38]: I think we should have a separate env bar for the TURNS port.

Comment (tbutter) [reply prosody/rootfs/defaults/conf.d/jitsi-meet.cfg.lua:38]: I think those usually are at the same port. Futhermore there is only one turncredentials_port option - i don't know if that would work with two ports.

**Comment by damencho** [comment]
I think we should use the external_services module as the debian package 
https://github.com/jitsi/jitsi-meet/blob/b2985934f7ea43b28c4fbc33bea855e74fa244ab/doc/debian/jitsi-meet-prosody/prosody.cfg.lua-jvb.example#L6
https://github.com/jitsi/jitsi-meet/blob/b2985934f7ea43b28c4fbc33bea855e74fa244ab/doc/debian/jitsi-meet-prosody/prosody.cfg.lua-jvb.example#L52

**Comment by tbutter** [comment]
> I think we should use the external_services module as the debian package
> https://github.com/jitsi/jitsi-meet/blob/b2985934f7ea43b28c4fbc33bea855e74fa244ab/doc/debian/jitsi-meet-prosody/prosody.cfg.lua-jvb.example#L6
> https://github.com/jitsi/jitsi-meet/blob/b2985934f7ea43b28c4fbc33bea855e74fa244ab/doc/debian/jitsi-meet-prosody/prosody.cfg.lua-jvb.example#L52

i can change it. what are the advantage of using that module?

**Comment by damencho** [comment]
Supports extdisco:2 https://github.com/jitsi/jitsi-meet/blob/b2985934f7ea43b28c4fbc33bea855e74fa244ab/resources/prosody-plugins/mod_external_services.lua#L236
And this module will be in prosody 0.12 by default.

File: prosody/rootfs/defaults/conf.d/jitsi-meet.cfg.lua (hunk 1 above)
Comment (tbutter) [reply prosody/rootfs/defaults/conf.d/jitsi-meet.cfg.lua]: By using the external_services module it was possible to set separate turn/turns servers

**Comment by tbutter** [comment]
@damencho thanks for the explanation. changed it to use that module.

File: env.example [hunk 2]
```diff
@@ -354,6 +354,13 @@ JIBRI_STRIP_DOMAIN_JID=muc
 # Directory for logs inside Jibri container
 JIBRI_LOGS_DIR=/config/logs
 
+# Configure an external TCP turn(s) server
```
Comment (saghul) [inline env.example:357]:
```suggestion
# Configure an external TURN server
```

File: prosody/rootfs/defaults/conf.d/jitsi-meet.cfg.lua [hunk 3]
```diff
@@ -91,6 +111,9 @@ VirtualHost "{{ .Env.XMPP_DOMAIN }}"
         "ping";
         "speakerstats";
         "conference_duration";
+        {{ if .Env.TURN_HOST }}
```
Comment (saghul) [inline prosody/rootfs/defaults/conf.d/jitsi-meet.cfg.lua:114]: also check  if TURNS_HOST is set?

File: env.example (hunk 2 above)
Comment (tbutter) [reply env.example]: thanks - fixed it

File: prosody/rootfs/defaults/conf.d/jitsi-meet.cfg.lua (hunk 3 above)
Comment (tbutter) [reply prosody/rootfs/defaults/conf.d/jitsi-meet.cfg.lua]: thanks - fixed it

**Review by saghul — APPROVED** [review]
LGTM. @damencho can you also give it a quick read please?

**Comment by seuffert** [comment]
i was confused and surprised that this was missing in the current implementation, thanks @tbutter ! i am currently using this PR without problems.

**Comment by damencho** [comment]
> LGTM. @damencho can you also give it a quick read please?

LGTM

**Comment by mayconmesquita** [comment]
@tbutter great implementation!
Is it possible to set a username, or only credential/pass?

**Comment by tbutter** [comment]
@mayconmesquita credentials is the secret for computing temporary passwords for arbitrary usernames as defined in https://datatracker.ietf.org/doc/html/draft-uberti-behave-turn-rest-00 section 2.2.

I don't know if jitsi supports setting a static username/password, but I wouldn't recommend it in any case. The username/password is transmitted to the users and should therefore only be temporary valid.

**Comment by mayconmesquita** [comment]
@tbutter Now I understand why there is no username. Thanks for the response!

**Pull request merged**